atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java

```diff
@@ -76,6 +77,40 @@ public void testRenderJava8OptionalsWhenRequested() throws IOException {
                         containsString("import java.util.Optional")));
     }
 
+    @Test
+    public void testIgnoreTableNameLengthFlag() throws IOException {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        schema.ignoreTableNameLengthChecks();
+        String longTableName = String.join("", Collections.nCopies(100, "x"));
+        TableReference tableRef = TableReference.createWithEmptyNamespace(longTableName);
+        schema.addTableDefinition(longTableName, getSimpleTableDefinition(tableRef));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testLongTableNameLengthFailsCassandra() throws IOException {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String longTableName = String.join("", Collections.nCopies(49, "x"));
+        TableReference tableRef = TableReference.createWithEmptyNamespace(longTableName);
+        schema.addTableDefinition(longTableName, getSimpleTableDefinition(tableRef));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testLongTableNameLengthFailsPostgres() throws IOException {
+        Schema schema = new Schema("Table", TEST_PACKAGE, Namespace.EMPTY_NAMESPACE);
+        String longTableName = String.join("", Collections.nCopies(64, "x"));
+        TableReference tableRef = TableReference.createWithEmptyNamespace(longTableName);
+        schema.addTableDefinition(longTableName, getSimpleTableDefinition(tableRef));
+    }
+
+    @Test(expected = IllegalArgumentException.class)
```

Comment:
We can check the message in the exception too, eg: https://github.com/palantir/atlasdb/blob/6fc0584daff813af50931b618981308e217294e6/atlasdb-tests-shared/src/main/java/com/palantir/atlasdb/transaction/impl/TransactionManagerTest.java#L62-L67. But wouldnt push hard on that.